iot_config/src/session_key.rs

```diff
@@ -0,0 +1,184 @@
+use crate::lora_field::DevAddrField;
+use futures::stream::{Stream, StreamExt};
+use helium_proto::services::iot_config::{
+    ActionV1, SessionKeyFilterStreamResV1, SessionKeyFilterV1,
+};
+use sqlx::{postgres::PgRow, FromRow, Row};
+use tokio::sync::broadcast::Sender;
+
+#[derive(Debug)]
+pub struct SessionKeyFilter {
+    pub oui: u64,
+    pub devaddr: DevAddrField,
+    pub session_key: String,
+}
+
+impl FromRow<'_, PgRow> for SessionKeyFilter {
+    fn from_row(row: &PgRow) -> sqlx::Result<Self> {
+        Ok(Self {
+            oui: row.try_get::<i64, &str>("oui")? as u64,
+            devaddr: row.try_get::<i64, &str>("devaddr")?.into(),
+            session_key: row.try_get::<String, &str>("session_key")?,
+        })
+    }
+}
+
+pub async fn list_stream<'a>(
+    db: impl sqlx::PgExecutor<'a> + 'a,
+) -> impl Stream<Item = SessionKeyFilter> + 'a {
+    sqlx::query_as::<_, SessionKeyFilter>("select * from session_key_filters")
+        .fetch(db)
+        .filter_map(|filter| async move { filter.ok() })
+}
+
+pub async fn list_for_oui(
+    oui: u64,
```

Comment:
as you need the oui as an i64 maybe enforce that here in the function sig and have the callers perform any cast.  Likewise for the other instance of same

Reply:
my goal is to wrap all of these "db wrapper" functions that wrap sqlx queries in a trait that can be tested with non-postgres impls for testing, in which case i'd keep this as-is since the i64 is a constraint of postgres only and the rust and proto definitions of these types all expect an unsigned integer